src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -295,11 +262,24 @@ public void clear(String feedId) {
   }
 
   /**
-   * Removes the latest added trip pattern from the cache. This should be done when removing the
-   * trip times from the timetable the trip has been added to.
+   * Removes previous trip-update from buffer if there is an update with given trip on service date
+   *
+   * @param serviceDate service date
+   * @return true if a previously added trip was removed
    */
-  public void removeLastAddedTripPattern(FeedScopedId feedScopedTripId, LocalDate serviceDate) {
-    realtimeAddedTripPattern.remove(new TripIdAndServiceDate(feedScopedTripId, serviceDate));
+  public boolean removePreviousRealtimeUpdate(FeedScopedId tripId, LocalDate serviceDate) {
```

Comment:
Is it accurate to say that only the following changes were made to method `removePreviousRealtimeUpdate`:
- Method was made public and pushed down from SiriTimetableSnapshotSource into TimetableSnapshot
- tripId is passed in instead of trip, because only trip.tripId is used by method
- all method calls qualified like  `buffer.method()` changed to `method()`

And these changes were made primarily to allow the method to be used from both SiriTimetableSnapshotSource and plain (GTFS) TimetableSnapshotSource?

I know this has been discussed before, I just want to confirm my understanding.

Reply:
Yes this is correct.